internal/rcache/recent_list.go

```diff
@@ -0,0 +1,101 @@
+package rcache
+
+import (
+	"context"
+	"fmt"
+	"unicode/utf8"
+
+	"github.com/gomodule/redigo/redis"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+)
+
+// RecentList holds the most recently inserted items, discarding older ones if the total item count goes over the configured size.
```

Comment:
Nitpick: I think "FIFO" (first in, first out) is more commonly used to describe the behavior here. "Recent" always makes me think in terms of dates and timestamps.

```suggestion
// FIFOList holds the most recently inserted items, discarding older ones if the total item count goes over the configured size.
```

Reply:
Makes sense, applying